src/components/InventoryTable/hooks/useInventoryExport/useExportDownloadCheck.js

```diff
@@ -0,0 +1,58 @@
+import { useRef, useState, useEffect, useCallback } from 'react';
+import { DOWNLOAD_CHECK_INTERVAL } from './constants';
+import useExportApi from './useExportApi';
+
+const useExportDownloadCheck = ({ onDownloadAvailable, onError }) => {
+  const { list } = useExportApi();
+  const checking = useRef(false);
+  const [checkForDownload, setCheckForDownload] = useState();
+
+  const checkDownloads = useCallback(
+    async (checkForExportId) => {
+      checking.current = true;
+      try {
+        const exportsToDownload = await list();
+        const exportToDownload = exportsToDownload.data.find(
+          ({ id, status }) => id === checkForExportId && status === 'complete'
+        );
+
+        if (exportToDownload) {
+          setCheckForDownload(undefined);
+          checking.current = false;
+
+          await onDownloadAvailable?.(exportToDownload);
+        } else {
+          checking.current = false;
+          console.log('No download for ', checkForExportId);
+        }
+      } catch (error) {
+        checking.current = false;
+        console.error(error);
+        setCheckForDownload(undefined);
+        onError?.(error);
+      }
+    },
+    [list, onDownloadAvailable, checking]
+  );
+
+  useEffect(() => {
+    if (checkForDownload) {
+      const downloadCheck = setInterval(async () => {
```

Comment:
Just a note: [Server side events](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events) would be probably perfect for this need, instead of polling the server. But, I know that is already late :)

Reply:
I could be wrong, but I think that server side events would only work if the server would sent events.

Reply:
Also, eventually, hopefully, 🤞, this check will go away and the Export service itself will trigger a notification for the download (via websockets).